service/src/main/kotlin/io/provenance/api/frameworks/provenance/ProvenanceService.kt

```diff
@@ -169,7 +191,73 @@ class ProvenanceService : Provenance {
             )
         }.txResponse.toTxResponse()
 
-    fun tryAction(config: ProvenanceConfig, signer: Signer, action: (pbClient: PbClient, account: Auth.BaseAccount, offset: Int) -> ServiceOuterClass.BroadcastTxResponse): ServiceOuterClass.BroadcastTxResponse {
+    override fun executeValidationOracleTransaction(
```

Comment:
This class is going to quickly become bloated. I'd recommend moving this out into a smart contract service perhaps?